osu.Game.Rulesets.Osu/Objects/Spinner.cs

```diff
@@ -93,6 +98,16 @@ protected override void CreateNestedHitObjects(CancellationToken cancellationTok
                     ? new SpinnerTick { StartTime = startTime, SpinnerDuration = Duration }
                     : new SpinnerBonusTick { StartTime = startTime, SpinnerDuration = Duration, Samples = new[] { CreateHitSampleInfo("spinnerbonus") } });
             }
+
+            // Add ticks that give HP across the whole spinner duration
+            int maxSpinsForHp = (int)(hp_ticks_per_minute * Duration / 60000);
```

Comment:
what is an HP tick? and why is it per minute?

Reply:
HP tick is a small bonus tick that only gives HP and doesn't affect scoring (to account for after-bonus-cap spinner ticks in stable/pre-spinner-changes).

It is per-minute as that's the way spinning speed is generally measured, and considering this change is meant to compensate for the difference of how bonus ticks were handled previously in stable (where they are awarded up to the end of the spinner object), it felt like a reasonable metric.

If this is decided to be a valid direction to go forward I'm open to exploring other suggestions though.